pkg/devicemapper/devmapper_wrapper.go

```diff
@@ -188,8 +188,11 @@ func dmTaskGetDepsFct(task *CDmTask) *Deps {
 		Count:  uint32(Cdeps.count),
 		Filler: uint32(Cdeps.filler),
 	}
-	for _, device := range Cdeps.device {
-		deps.Device = append(deps.Device, (uint64)(device))
+	// this gem from rsc
+	// https://groups.google.com/d/msg/golang-dev/iybwRUoA1Vg/mAvIrl9rBwAJ
+	devices := (*[1 << 30]uint64)(unsafe.Pointer(uintptr(unsafe.Pointer(Cdeps)) + unsafe.Sizeof(*Cdeps)))[:Cdeps.count]
```

Comment:
it just shifts the bit to make a large limit on the size of the slice returned. not magic. :-)